src/app/static/src/app/dataExportService.ts

```diff
@@ -10,17 +10,24 @@ export interface ExportService {
 export class DataExportService implements ExportService {
     private readonly _data: DownloadIndicatorDataset
     private readonly _filename: string
+    private readonly _options: any
 
     private readonly _workbook: XLSX.WorkBook;
 
     constructor(payload: DownloadIndicatorPayload) {
         this._filename = payload.filename
         this._data = payload.data
         this._workbook = XLSX.utils.book_new();
+        this._options = payload.options;
     }
 
     addFilteredData = (): ExportService => {
-        const worksheet = XLSX.utils.json_to_sheet(this._data.filteredData);
+        let worksheet;
+        if (this._options) {
+            worksheet = XLSX.utils.json_to_sheet(this._data.filteredData, this._options);
```

Comment:
So this will only include columns for the specified headers in `options`, not any other fields which may be in the data?

Reply:
hmm i think so yh, but these are actually all the fields in the data, it just lets me reorder them!